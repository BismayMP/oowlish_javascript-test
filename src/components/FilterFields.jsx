import React, { useState } from 'react';

const FilterFields = ({ name, UPIN, onFilterChange }) => {

    return (
        <>
                <input
                    type="text"
                    name='name'
                    placeholder='Name'
                    onChange={onFilterChange}
                    value={name}>
                </input>
                <input
                    type="text"
                    name='UPIN'
                    placeholder='UPIN'
                    onChange={onFilterChange}
                    value={UPIN}
                >
                </input>
        </>
    )
}

export default FilterFields;