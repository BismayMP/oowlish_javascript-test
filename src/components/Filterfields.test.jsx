
import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import FilterFields from './FilterFields';

afterEach(cleanup);
test('Component render', () => {
    expect(renderer.create(
        <FilterFields
            name='mock name'
            UPIN='1234' />).toJSON()).toMatchSnapshot()

})

const mockOnChange = jest.fn()

test('test name input', () => {
    const { getByDisplayValue } = render(<FilterFields
        name='mock name'
        UPIN='1234'
        onFilterChange={mockOnChange} />)
    expect(getByDisplayValue('1234')).toBeTruthy();
})

test('test UPIN input', () => {
    const { getByDisplayValue } = render(<FilterFields
        name='mock name'
        UPIN='1234'
        onFilterChange={mockOnChange} />)
    expect(getByDisplayValue('1234')).toBeTruthy();
})