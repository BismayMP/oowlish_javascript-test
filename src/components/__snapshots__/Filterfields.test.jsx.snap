// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render 1`] = `
Array [
  <input
    name="name"
    placeholder="Name"
    type="text"
    value="mock name"
  />,
  <input
    name="UPIN"
    placeholder="UPIN"
    type="text"
    value="1234"
  />,
]
`;
